pkg/zpdb/metrics.go

```diff
@@ -0,0 +1,29 @@
+package zpdb
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type Metrics struct {
+	ConfigurationsObserved *prometheus.CounterVec
+	EvictionRequests       *prometheus.CounterVec
+	InFlightRequests       *prometheus.GaugeVec
+}
+
+func NewMetrics(reg prometheus.Registerer) *Metrics {
+	return &Metrics{
+		ConfigurationsObserved: promauto.With(reg).NewCounterVec(prometheus.CounterOpts{
+			Name: "rollout_operator_zpdb_configurations_observed_total",
+			Help: "Number of zpdb configurations observed by the configuration controller.",
+		}, []string{"result", "name"}),
```

Comment:
Including the ZPDB name as a label is going to increase the cardinality significantly - is there a reason behind including it?